sdk/python/feast/client.py

```diff
@@ -311,8 +311,8 @@ def list_projects(self) -> List[str]:
             List of project names
 
         """
-        self._connect_core()
-        response = self._core_service_stub.ListProjects(
+        self._connect_core()  # ensures self._core_service_stub is defined
+        response = self._core_service_stub.ListProjects(  # type: ignore
```

Comment:
Why are you using `type: ignore` so frequently? Can we set these types?

Reply:
`_core_service_stub` has type `Optional[CoreServiceStub]`, which Mypy throws error when we call it as it could be `None`. However `_connect_core()` guarantees that `_core_service_stub` is non-null, but Mypy could not detect.

Reply:
I wrote it in the commit message here too: https://github.com/feast-dev/feast/pull/749/commits/ffbe15bf5a95b658e388387c7f4220230a82b2fb